# 🌍 DeMark-World - Google Colab Setup

**Universal AI Video Watermark Removal** - Works with Sora, Runway, Veo, Pika, Kling, Luma and more!

**Important:** Make sure you have GPU enabled in Runtime > Change runtime type > Hardware accelerator > GPU (T4 or better)

## Features:
- 🎯 Universal watermark detection (YOLOv11)
- ⚡ LAMA model for fast removal
- 🎬 E2FGVI_HQ for time-consistent (flicker-free) results
- 🚀 Modern stack (PyTorch 2.5+, NumPy 2.x)

**Repo**: https://github.com/linkedlist771/DeMark-World

---

## Step 1: Install System Dependencies

In [7]:
import os
from google.colab import drive

# ── Mount Google Drive FIRST so the permission prompt happens immediately ────
# This way you can approve access and then walk away — no surprise prompts later.
print("📂 Mounting Google Drive...")
drive.mount('/content/drive')
print("✅ Drive mounted\n")

# ── Install FFmpeg ────────────────────────────────────────────────────────────
!apt-get update -qq
!apt-get install -y ffmpeg
!ffmpeg -version | head -n 1

# ── Install uv ───────────────────────────────────────────────────────────────
!curl -LsSf https://astral.sh/uv/install.sh | sh
os.environ['PATH'] = f"/root/.cargo/bin:{os.environ['PATH']}"
!uv --version

📂 Mounting Google Drive...
Mounted at /content/drive
✅ Drive mounted

W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
ffmpeg is already the newest version (7:4.4.2-0ubuntu0.22.04.1).
0 upgraded, 0 newly installed, 0 to remove and 45 not upgraded.
ffmpeg version 4.4.2-0ubuntu0.22.04.1 Copyright (c) 2000-2021 the FFmpeg developers
downloading uv 0.10.2 x86_64-unknown-linux-gnu
no checksums to verify
installing to /usr/local/bin
  uv
  uvx
everything's installed!
uv 0.10.2


## Step 2: Clone DeMark-World Repository

In [8]:
# Clone the repository
!git clone https://github.com/linkedlist771/DeMark-World.git
%cd DeMark-World

Cloning into 'DeMark-World'...
remote: Enumerating objects: 376, done.
remote: Counting objects: 100% (376/376), done.
remote: Compressing objects: 100% (309/309), done.
remote: Total 376 (delta 65), reused 357 (delta 51), pack-reused 0 (from 0)
Receiving objects: 100% (376/376), 12.04 MiB | 17.95 MiB/s, done.
Resolving deltas: 100% (65/65), done.
/content/DeMark-World/DeMark-World


## Step 3: Pre-install build tools using global uv + venv python

In [9]:
# 1. Clean venv
!rm -rf .venv

# 2. Create fresh venv
!uv venv --clear

# 3. Install required build plugins + old setuptools (covers CLIP too)
!uv pip install --python .venv/bin/python \
  hatchling \
  uv-dynamic-versioning \
  editables \
  "setuptools<70" \
  wheel \
  build

# 4. Sync
!uv sync --no-build-isolation

Using CPython 3.12.12 interpreter at: /usr/bin/python3
Creating virtual environment at: .venv
Activate with: source .venv/bin/activate
Resolved 15 packages in 59ms
Installed 15 packages in 22ms
 + build==1.4.0
 + dunamai==1.25.0
 + editables==0.5
 + hatchling==1.28.0
 + jinja2==3.1.6
 + markupsafe==3.0.3
 + packaging==26.0
 + pathspec==1.0.4
 + pluggy==1.6.0
 + pyproject-hooks==1.2.0
 + setuptools==69.5.1
 + tomlkit==0.14.0
 + trove-classifiers==2026.1.14.14
 + uv-dynamic-versioning==0.13.0
 + wheel==0.46.3
Resolved 209 packages in 1ms
Uninstalled 2 packages in 4ms
Installed 201 packages in 821ms
Prepared 1 package without build isolation in 707ms
Uninstalled 10 packages in 7ms
Installed 1 package in 1ms
 + addict==2.4.0
 + aiofiles==25.1.0
 + aiosqlite==0.21.0
 + altair==5.5.0
 + annotated-types==0.7.0
 + antlr4-python3-runtime==4.9.3
 + anyio==4.11.0
 + argon2-cffi==25.1.0
 + argon2-cffi-bindings==25.1.0
 + arrow==1.4.0
 + asttokens==3.0.1
 + async-lru==2.0.5
 + attrs==25.4.0
 + babe

## Step 4: Verify Installation

In [10]:
# Activate UV environment and verify packages
import sys
sys.path.insert(0, '/content/DeMark-World/.venv/lib/python3.12/site-packages')

import torch
import numpy as np
import torchvision

print("="*60)
print("ENVIRONMENT VERIFICATION")
print("="*60)
print(f"Python: {sys.version.split()[0]}")
print(f"PyTorch: {torch.__version__}")
print(f"Torchvision: {torchvision.__version__}")
print(f"NumPy: {np.__version__}")
print(f"CUDA available: {torch.cuda.is_available()}")

if torch.cuda.is_available():
    print(f"GPU: {torch.cuda.get_device_name(0)}")

# Check key packages
try:
    import cv2
    print(f"✅ OpenCV: {cv2.__version__}")
except:
    print("❌ OpenCV: Not available")

try:
    import mmcv
    print(f"✅ MMCV: {mmcv.__version__}")
except:
    print("❌ MMCV: Not available (will be installed on first run)")

try:
    from ultralytics import YOLO
    print("✅ Ultralytics: Available")
except:
    print("❌ Ultralytics: Not available")

try:
    import transformers
    print(f"✅ Transformers: {transformers.__version__}")
except:
    print("❌ Transformers: Not available")

try:
    import diffusers
    print(f"✅ Diffusers: {diffusers.__version__}")
except:
    print("❌ Diffusers: Not available")

print("="*60)
print("\n🎉 All checks passed! Ready to process videos.")

ENVIRONMENT VERIFICATION
Python: 3.12.12
PyTorch: 2.9.1+cu128
Torchvision: 0.24.1+cu128
NumPy: 2.0.2
CUDA available: True
GPU: Tesla T4
✅ OpenCV: 4.12.0
✅ MMCV: 1.7.2
✅ Ultralytics: Available
✅ Transformers: 4.57.1
✅ Diffusers: 0.35.2

🎉 All checks passed! Ready to process videos.


## Step 6: Initialize DeMark-World

In [11]:
# Set up environment
%cd /content/DeMark-World
import sys
sys.path.insert(0, '/content/DeMark-World/src')

# Create output directories
!mkdir -p outputs resources

# Import DeMark-World
print("Initializing DeMark-World...")
try:
    from demark_world.core import DeMarkWorld
    from demark_world.schemas import CleanerType

    print("✅ DeMark-World imported successfully!")
    print("\nℹ️  Models will be downloaded automatically on first use.")
    print("   This may take a few minutes depending on your connection.")
except Exception as e:
    print(f"❌ Error: {e}")
    import traceback
    traceback.print_exc()

/content/DeMark-World
Initializing DeMark-World...
✅ DeMark-World imported successfully!

ℹ️  Models will be downloaded automatically on first use.
   This may take a few minutes depending on your connection.


## Step 6.5 (Alternative): 🤖 Automated Batch Processing

**Auto-monitor Google Drive folders and process videos automatically!**

This replaces Steps 7-10 with a fully automated workflow:
- Drop videos into `input` folder in Google Drive
- Script automatically processes them with LAMA (fast)
- Outputs saved to `output` folder
- Processed inputs moved to `processed` folder

Perfect for continuous background processing!

In [ ]:
#@title 🚀 Launch Auto-Processing { display-mode: "form" }

#@markdown ---
#@markdown ### ⚙️ Configuration
enable_auto_processing = True  #@param {type:"boolean"}
check_interval_seconds = 10    #@param {type:"slider", min:5, max:60, step:5}
drive_base_folder = "Figuro/sora-watermark-remover" #@param {type:"string"}
use_lama = True                #@param {type:"boolean"}
extra_output_path = "Figuro/video-upscale/input"         #@param {type:"string"}

import torch, time, os, sys, shutil, threading
from pathlib import Path
from datetime import datetime
from demark_world.core import DeMarkWorld
from demark_world.schemas import CleanerType
from google.colab import drive

# ── Kill any previous monitor loop ──────────────────────────────────────────
_STOP_EVENT = getattr(sys.modules[__name__], '_STOP_EVENT', None)
if _STOP_EVENT is not None:
    print("⏹️  Stopping previous monitor...")
    _STOP_EVENT.set()
    time.sleep(2)
_STOP_EVENT = threading.Event()
sys.modules[__name__]._STOP_EVENT = _STOP_EVENT

# ── 1. Drive (already mounted in Step 1) ─────────────────────────────────────
if not os.path.isdir('/content/drive/MyDrive') and not os.path.isdir('/content/drive/Shareddrives'):
    raise RuntimeError("❌ Drive not mounted — run Step 1 first.")
print("✅ Drive already mounted")

# ── 2. Auto-detect correct Shared Drive path ─────────────────────────────────
def find_drive_base(subfolder: str) -> str:
    candidates = [
        f"/content/drive/Shareddrives/{subfolder}",
        f"/content/drive/Shared drives/{subfolder}",
        f"/content/drive/MyDrive/{subfolder}",
    ]
    drive_root = "/content/drive"
    try:
        for top in os.listdir(drive_root):
            top_path = os.path.join(drive_root, top)
            if os.path.isdir(top_path):
                candidate = os.path.join(top_path, subfolder)
                if candidate not in candidates:
                    candidates.append(candidate)
    except Exception:
        pass

    print("\n🔍 Searching for drive path...")
    for c in candidates:
        parent = str(Path(c).parent)
        if os.path.isdir(parent):
            print(f"  ✅ Found: {c}")
            return c
        else:
            print(f"  ✗  {c}")

    print("\n📁 Available paths under /content/drive:")
    try:
        for top in os.listdir(drive_root):
            top_path = os.path.join(drive_root, top)
            print(f"  {top}/")
            try:
                for sub in os.listdir(top_path)[:5]:
                    print(f"    {sub}/")
            except Exception:
                pass
    except Exception as e:
        print(f"  Could not list: {e}")

    raise RuntimeError(
        f"Could not find '{subfolder}' under /content/drive.\n"
        "Check the drive_base_folder setting and that the Shared Drive is accessible."
    )

DRIVE_BASE = find_drive_base(drive_base_folder)
folders = {
    "input":     f"{DRIVE_BASE}/input",
    "processed": f"{DRIVE_BASE}/processed",
    "output":    f"{DRIVE_BASE}/output",
}
for p in folders.values():
    os.makedirs(p, exist_ok=True)

# ── Extra output folder ───────────────────────────────────────────────────────
_extra = extra_output_path.strip()
if _extra:
    # If user gave a bare subfolder (no /content/drive prefix), auto-resolve it
    if not _extra.startswith('/'):
        EXTRA_DIR = str(Path(DRIVE_BASE).parent.parent / _extra)
    else:
        EXTRA_DIR = _extra
    os.makedirs(EXTRA_DIR, exist_ok=True)
else:
    EXTRA_DIR = None

print("\n" + "=" * 60)
print("📂 FOLDER STRUCTURE READY")
print("=" * 60)
for k, v in folders.items():
    print(f"  {k:<10} {v}")
if EXTRA_DIR:
    print(f"  {'extra':<10} {EXTRA_DIR}")
print("=" * 60)

# ── 3. GPU / CPU info ────────────────────────────────────────────────────────
print("\n🖥️  COMPUTE INFO")
print("=" * 40)
if torch.cuda.is_available():
    gpu       = torch.cuda.get_device_properties(0)
    mem_total = gpu.total_memory / 1024**3
    mem_used  = torch.cuda.memory_allocated(0) / 1024**3
    print(f"  ✅ GPU : {gpu.name}")
    print(f"     VRAM: {mem_used:.1f} GB used / {mem_total:.1f} GB total")
    print(f"     CUDA: {torch.version.cuda}")
else:
    print("  ⚠️  No GPU — running on CPU (will be slow!)")
print("=" * 40)

# ── 4. Model init ────────────────────────────────────────────────────────────
cleaner_type = CleanerType.LAMA if use_lama else CleanerType.E2FGVI_HQ
model_label  = "LAMA" if use_lama else "E2FGVI_HQ"
print(f"\n🔧 Initialising {model_label}…", flush=True)
demark = DeMarkWorld(cleaner_type=cleaner_type)
print(f"✅ {model_label} ready!\n", flush=True)

# ── 5. Helpers ───────────────────────────────────────────────────────────────
SUPPORTED   = ('.mp4', '.mov', '.avi', '.mkv', '.webm')
_PROCESSING = set()
_PROC_LOCK  = threading.Lock()

def scan_input() -> list[str]:
    try:
        os.system("sync")
        all_files = sorted(
            f for f in os.listdir(folders["input"])
            if f.lower().endswith(SUPPORTED)
        )
        with _PROC_LOCK:
            return [f for f in all_files if f not in _PROCESSING]
    except FileNotFoundError:
        print("⚠️  Input folder missing — check Drive mount", flush=True)
        return []
    except Exception as e:
        print(f"⚠️  Scan error: {e}", flush=True)
        return []

def process_file(filename: str):
    with _PROC_LOCK:
        if filename in _PROCESSING:
            return
        _PROCESSING.add(filename)

    input_path  = Path(folders["input"]) / filename
    ts          = datetime.now().strftime("%Y%m%d_%H%M%S")

    safe_stem   = Path(filename).stem.replace(" ", "_").replace("(","").replace(")","")
    safe_name   = f"{safe_stem}{Path(filename).suffix}"
    temp_path   = Path("/tmp") / safe_name
    out_name    = f"{ts}_{safe_stem}_cleaned.mp4"
    output_path = Path(folders["output"]) / out_name

    print(f"\n{'─'*60}", flush=True)
    print(f"▶  {filename}", flush=True)
    print(f"   → {out_name}", flush=True)
    print(f"   Started : {datetime.now().strftime('%H:%M:%S')}", flush=True)
    print(f"{'─'*60}", flush=True)

    t0 = time.time()
    try:
        shutil.copy2(str(input_path), str(temp_path))

        import tqdm as tqdm_module
        orig_file = getattr(tqdm_module.tqdm, 'file', None)
        tqdm_module.tqdm.file = sys.stdout

        demark.run(temp_path, output_path)

        if orig_file:
            tqdm_module.tqdm.file = orig_file

        temp_path.unlink(missing_ok=True)

        mins, secs = divmod(int(time.time() - t0), 60)
        print(f"\n✅ Done in {mins}m {secs}s  →  {out_name}", flush=True)

        # ── Copy to extra output if configured ───────────────────────────
        if EXTRA_DIR:
            extra_path = Path(EXTRA_DIR) / out_name
            shutil.copy2(str(output_path), str(extra_path))
            print(f"📤 Extra copy →  {extra_path}", flush=True)

        archived = Path(folders["processed"]) / f"{ts}_{safe_name}"
        shutil.move(str(input_path), str(archived))
        print(f"📦 Archived  →  processed/{archived.name}\n", flush=True)

    except Exception as e:
        mins, secs = divmod(int(time.time() - t0), 60)
        print(f"\n❌ Failed after {mins}m {secs}s: {e}", flush=True)
        temp_path.unlink(missing_ok=True)
        err_path = Path(folders["processed"]) / f"{ts}_ERROR_{safe_name}"
        try:
            shutil.move(str(input_path), str(err_path))
            print(f"🗂️  Moved to  processed/{err_path.name}\n", flush=True)
        except Exception as mv_err:
            print(f"   (could not move: {mv_err})\n", flush=True)
    finally:
        with _PROC_LOCK:
            _PROCESSING.discard(filename)

# ── 6. Main loop (foreground) ────────────────────────────────────────────────
if not enable_auto_processing:
    print("⏸️  Disabled. Enable in configuration above.")
else:
    print(f"🤖 Watching  {folders['input']}")
    print(f"   Interval : {check_interval_seconds}s  |  Model: {model_label}")
    if EXTRA_DIR:
        print(f"   Extra out: {EXTRA_DIR}")
    print("   Interrupt kernel (■) to stop.\n", flush=True)

    heartbeat = 0
    while not _STOP_EVENT.is_set():
        try:
            pending = scan_input()

            if pending:
                print(f"\n🎬 {len(pending)} queued: {', '.join(pending)}", flush=True)
                for fn in pending:
                    if _STOP_EVENT.is_set():
                        break
                    process_file(fn)
                print("✔  Queue empty — resuming watch…\n", flush=True)
                heartbeat = 0
            else:
                heartbeat += 1
                print("·", end=("\n" if heartbeat % 30 == 0 else ""), flush=True)

            time.sleep(check_interval_seconds)

        except KeyboardInterrupt:
            print("\n\n🛑 Stopped by user.")
            break
        except Exception as e:
            print(f"\n⚠️  Loop error: {e} — retrying in 30s…", flush=True)
            time.sleep(30)

⏹️  Stopping previous monitor...
✅ Drive already mounted

🔍 Searching for drive path...
  ✅ Found: /content/drive/Shareddrives/Figuro/sora-watermark-remover

📂 FOLDER STRUCTURE READY
  input      /content/drive/Shareddrives/Figuro/sora-watermark-remover/input
  processed  /content/drive/Shareddrives/Figuro/sora-watermark-remover/processed
  output     /content/drive/Shareddrives/Figuro/sora-watermark-remover/output
  extra      /content/drive/Shareddrives/Figuro/video-upscale/input

🖥️  COMPUTE INFO
  ✅ GPU : Tesla T4
     VRAM: 0.0 GB used / 14.6 GB total
     CUDA: 12.8

🔧 Initialising LAMA…


2026-02-13 11:03:10.669 | DEBUG    | src.demark_world.utils.download_utils:ensure_model_downloaded:17 - Downloading model from https://github.com/linkedlist771/DeMark-World/releases/download/V0.0.1/watermark-best.pt
Downloading: 100%|██████████| 114M/114M [00:02<00:00, 48.7MB/s]
2026-02-13 11:03:13.574 | SUCCESS  | src.demark_world.utils.download_utils:ensure_model_downloaded:33 - ✓ Model downloaded: /content/DeMark-World/resources/checkpoint/watermark-best.pt
2026-02-13 11:03:13.578 | DEBUG    | src.demark_world.watermark_detector:__init__:20 - Begin to load yolo water mark detet model.
2026-02-13 11:03:14.674 | DEBUG    | src.demark_world.watermark_detector:__init__:24 - Yolo water mark detet model loaded from /content/DeMark-World/resources/checkpoint/watermark-best.pt.
2026-02-13 11:03:14.694 | INFO     | src.demark_world.cleaner.lama_cleaner:__init__:25 - lama not found in /root/.cache, try to downloading
2026-02-13 11:03:14.694 | INFO     | src.demark_world.iopaint.download:cli_d

✅ LAMA ready!

🤖 Watching  /content/drive/Shareddrives/Figuro/sora-watermark-remover/input
   Interval : 10s  |  Model: LAMA
   Extra out: /content/drive/Shareddrives/Figuro/video-upscale/input
   Interrupt kernel (■) to stop.


🎬 1 queued: Copy of 20260209_1520_01kh0ptyjvecnawm6n864z55de.mp4

────────────────────────────────────────────────────────────
▶  Copy of 20260209_1520_01kh0ptyjvecnawm6n864z55de.mp4
   → 20260213_110317_Copy_of_20260209_1520_01kh0ptyjvecnawm6n864z55de_cleaned.mp4
   Started : 11:03:17
────────────────────────────────────────────────────────────


2026-02-13 11:03:18.093 | DEBUG    | demark_world.core:run:123 - total frames: 300, fps: 30.0, width: 704, height: 1280
Detect watermarks: 100%|██████████| 300/300 [00:48<00:00,  6.20it/s]
2026-02-13 11:04:06.521 | DEBUG    | demark_world.core:run:151 - detect missed frames: [0, 1, 114, 228]
2026-02-13 11:04:06.552 | DEBUG    | demark_world.core:run:179 - Filled missed frame 0 with bbox:
 (55, 51, 225, 121)
2026-02-13 11:04:06.553 | DEBUG    | demark_world.core:run:179 - Filled missed frame 1 with bbox:
 (55, 51, 225, 121)
2026-02-13 11:04:06.553 | DEBUG    | demark_world.core:run:179 - Filled missed frame 114 with bbox:
 (55, 51, 225, 121)
2026-02-13 11:04:06.554 | DEBUG    | demark_world.core:run:179 - Filled missed frame 228 with bbox:
 (53, 1154, 225, 1223)
Remove watermarks: 100%|██████████| 300/300 [02:05<00:00,  2.39it/s]
2026-02-13 11:06:19.235 | INFO     | demark_world.core:merge_audio_track:325 - Merging audio track...
2026-02-13 11:06:19.683 | INFO     | demark_world.core:me


✅ Done in 3m 2s  →  20260213_110317_Copy_of_20260209_1520_01kh0ptyjvecnawm6n864z55de_cleaned.mp4
📤 Extra copy →  /content/drive/Shareddrives/Figuro/video-upscale/input/20260213_110317_Copy_of_20260209_1520_01kh0ptyjvecnawm6n864z55de_cleaned.mp4
📦 Archived  →  processed/20260213_110317_Copy_of_20260209_1520_01kh0ptyjvecnawm6n864z55de.mp4

✔  Queue empty — resuming watch…

················

## Step 7: Upload Your Video (Manual Processing)

**⚠️ Skip this if you're using Step 6.5 (Automated Batch Processing)**

Upload any AI-generated video with watermarks (Sora, Runway, Veo, Pika, Kling, Luma, etc.)

In [ ]:
from google.colab import files
from pathlib import Path
import shutil

print("Please upload your video file:")
uploaded = files.upload()

# Get the uploaded filename
input_filename = list(uploaded.keys())[0]
input_path = Path(input_filename)

# Move to resources folder
shutil.move(input_filename, f"resources/{input_filename}")
input_video_path = Path(f"resources/{input_filename}")

print(f"\n✅ Uploaded: {input_filename}")
print(f"📁 Saved to: {input_video_path}")

## Step 8: Process Video - LAMA (Fast)

**LAMA**: Fast processing, good quality. Recommended for most use cases.

In [ ]:
from pathlib import Path
from demark_world.core import DeMarkWorld
from demark_world.schemas import CleanerType

# Set output path
output_video_path = Path("outputs/watermark_removed_lama.mp4")

print("🚀 Processing with LAMA model (fast mode)...")
print("⏱️  Typical speed: 2-5 seconds per second of video (T4 GPU)")
print("")

# Initialize and run
demark = DeMarkWorld(cleaner_type=CleanerType.LAMA)
demark.run(input_video_path, output_video_path)

print(f"\n✅ Done! Video saved to: {output_video_path}")
print("Proceed to Step 10 to download.")

## Step 9: Process Video - E2FGVI_HQ (High Quality)

**E2FGVI_HQ**: Time-consistent, flicker-free results. Slower but higher quality.

⚠️ **Skip this if you already ran LAMA above!** Only run this if you want higher quality.

In [ ]:
import os
from pathlib import Path
from demark_world.core import DeMarkWorld
from demark_world.schemas import CleanerType

# Optimize PyTorch memory management to reduce fragmentation
os.environ['PYTORCH_CUDA_ALLOC_CONF'] = 'expandable_segments:True'

# Clear any cached memory
import torch
torch.cuda.empty_cache()

# Set output path
output_video_path = Path("outputs/watermark_removed_e2fgvi.mp4")

print("🚀 Processing with E2FGVI_HQ model (high quality, flicker-free)...")
print("⏱️  Typical speed: 10-20 seconds per second of video (L4 GPU)")
print("⚠️  This will take longer but produces superior results")
print("")

# Initialize and run
demark = DeMarkWorld(cleaner_type=CleanerType.E2FGVI_HQ)
demark.run(input_video_path, output_video_path)

print(f"\n✅ Done! Video saved to: {output_video_path}")
print("Proceed to Step 10 to download.")

## Step 10: Download Processed Video

In [ ]:
from google.colab import files

# Download the output video
# Use the appropriate filename based on which method you used
output_file = str(output_video_path)  # Will use the last output_video_path from Step 8 or 9

print(f"Downloading: {output_file}")
files.download(output_file)
print("✅ Download started!")

## Step 11 (Optional): Manual Batch Processing

**⚠️ Skip this if you're using Step 6.5 (Automated)**

Process multiple videos at once - upload all at once and process in sequence.

In [ ]:
# Upload multiple videos
from google.colab import files
import shutil
from pathlib import Path

print("Upload multiple video files:")
uploaded_files = files.upload()

# Create batch input/output folders
batch_input = Path("batch_input")
batch_output = Path("batch_output")
batch_input.mkdir(exist_ok=True)
batch_output.mkdir(exist_ok=True)

# Move uploaded files
for filename in uploaded_files.keys():
    shutil.move(filename, batch_input / filename)

print(f"\n✅ {len(uploaded_files)} files uploaded and ready for processing")

In [ ]:
# Process all videos in batch
from pathlib import Path
from demark_world.core import DeMarkWorld
from demark_world.schemas import CleanerType
from tqdm import tqdm

batch_input = Path("batch_input")
batch_output = Path("batch_output")

# Get all video files
video_files = list(batch_input.glob("*.mp4")) + list(batch_input.glob("*.mov")) + list(batch_input.glob("*.avi"))

print(f"Processing {len(video_files)} videos...")
print("")

# Initialize model (LAMA for speed)
demark = DeMarkWorld(cleaner_type=CleanerType.LAMA)

# Process each video
for video_file in tqdm(video_files, desc="Processing videos"):
    output_file = batch_output / f"cleaned_{video_file.name}"
    print(f"\nProcessing: {video_file.name}")

    try:
        demark.run(video_file, output_file)
        print(f"✅ Completed: {output_file.name}")
    except Exception as e:
        print(f"❌ Error processing {video_file.name}: {e}")

print("\n" + "="*60)
print("✅ Batch processing complete!")
print("="*60)

In [ ]:
# Download all processed videos
from google.colab import files
from pathlib import Path

batch_output = Path("batch_output")

print("Downloading all processed videos...")
for video_file in batch_output.glob("*.mp4"):
    print(f"Downloading: {video_file.name}")
    files.download(str(video_file))

print("\n✅ All downloads started!")

## Tips & Performance Notes

### Model Selection:
- **LAMA**: Fast (~2-5 sec/video-sec), good for quick results and testing
- **E2FGVI_HQ**: Slower (~10-20 sec/video-sec) but produces flicker-free, time-consistent results

### Supported Platforms:
DeMark-World works with watermarks from:
- ✅ OpenAI Sora (v1, v2)
- ✅ Google Gemini / Veo (3.1)
- ✅ Runway Gen-3 / Gen-4
- ✅ Pika Labs
- ✅ Kling AI
- ✅ Luma Dream Machine
- ✅ Any AI video generator with consistent watermarks

### GPU Requirements:
- **Minimum**: Google Colab T4 (free tier)
- **Recommended**: T4 or better for faster processing
- **Note**: Enable GPU in Runtime > Change runtime type

### Common Issues:
1. **Out of Memory**: Try LAMA instead of E2FGVI_HQ
2. **Slow Processing**: Ensure GPU is enabled
3. **Model Download Fails**: Check internet connection, models auto-retry

---

**Project Repository**: https://github.com/linkedlist771/DeMark-World

**License**: MIT

**Citation**:
```
@software{demarkworld2025,
  author = {linkedlist771},
  title = {DeMark-World: Universal AI Video Watermark Removal},
  year = {2025},
  url = {https://github.com/linkedlist771/DeMark-World}
}
```